### DISCUSSION TITLE
Generic RFID tag support possible?

### DISCUSSION THREAD

**Discussion by danny6869** [OrcaSlicer/OrcaSlicer#623] [Ideas]
Is the AMS' RFID reader accessible from the slicer?  If so, then perhaps an open tag format can be developed to allow people to stick a programmable rfid tag on their non bambu filament spools, allowing the X1C AMS to autodetect and configure based on that?

I would be willing to jump in and help on that if this is at all doable?

**Comment by SoftFever** [comment]
Hi @danny6869 
If we are specifically talking about Bambulab's machines, there is unfortunately no way to do that without hacking into their firmware first...
But maybe we can bypass that by creating an independent IoT device to read and send RFID to the slicer.
Which can be used for all 3d printers.
Thoughts?

**Reply by danny6869** [reply]
That sounds like a decent idea to me.  How would the sensor/data need to be accessed by orca slicer?  I can give this some thought, although I am more of a software person than hardware, I could probably come up with something.

**Reply by atl285** [reply]
An RFID tag system has at least to be integrated in the printer management (e.g. Octoprint) or in the firmware (e.g. Klipper). Simply to know, what material is currently loaded / used on the printer, is the only - but not really an - effort. If the firmware also knows the material, it can check if the right material loaded and react on the loaded material (stop if wrong material is detected). The most sense it makes in combination with multi material systems. These have also to support this.

**Reply by SoftFever** [reply]
Integrating with firmware could be a bonus feature.
The most helpful part is that the slicer automatically reads the RFID info to choose the corresponding filament.
OrcaSlicer will access the filament info from a Raspberry Pi server(REST API). Raspberry Pi reads the sensor.
For Klipper machines, the same RPi can be used.
For other machines, a Raspberry Pi Zero W  should be sufficient.

**Reply by loaderup** [reply]
Could there be a way to have this work if you have multiple machine that are used in a print farm?

**Reply by danny6869** [reply]
I just happen to have a PiZero right here...Rest API...no problem.  I'll see if I can make time over the next week or two to prototype something usable

**Reply by atl285** [reply]
@SoftFever, I think it's only a first step to be able to read the type of filament from printer. 

For my workflow this does not applies really, because I select the filament based on the type of project and not on what's loaded by the printer. More important I think is, that the printer refuses to print if not the right / selected filament is loaded on print start. My printer is located in the basement, one floor below the desktop. At the moment mostly I send the file to the printer and then go down to the printer to load the right filament and start the print on the device or start it via mobile device. The RFID solution would allow to start the print from desktop and get notified from printer, if the wrong filament is loaded. So only then it is necessary to go down to the printer.

**Reply by chaugan** [reply]
> But maybe we can bypass that by creating an independent IoT device to read and send RFID to the slicer.

I really would like this. I already have my printer in Home Assistant with stats and everything, and it would be great to have some kind of IoT feature to populate the slicer with info from an RFID reader in the AMS.

**Comment by Ricky1966** [comment]
If you are interested, I am carrying out an Open Source project to build a sort of AMS, but free.
My intent would be to insert different values into Tag215 which would then be transmitted to Klipper using Spoolman.  If you're interested, I can explain what I'm working on.
The project is already quite advanced, but there are two of us and we can only collaborate via the web.
If you are interested, we are open to collaborating!
